assets/app/view/game/buy_trains.rb

```diff
@@ -162,6 +162,7 @@ def from_depot(depot_trains)
             price = variant[:price]
             president_assist, _fee = @game.president_assisted_buy(@corporation, train, price)
             price = @ability&.discounted_price(train, price) || price
+            price = @game.apply_discard_discount(train, price)
```

Comment:
I could use `respond_to` for this instead of putting it in game base, but that felt even ickier